sdk/eventhub/event-hubs/src/impl/amqpContextHelpers.ts

```diff
@@ -0,0 +1,283 @@
+import {
+  TokenCredential,
+  EventHubConnectionConfig,
+  SharedKeyCredential,
+  ConnectionConfig,
+  parseConnectionString,
+  EventHubConnectionStringModel
+} from "@azure/core-amqp";
+
+import { ConnectionContext } from "../connectionContext";
+import { PartitionProperties, EventHubProperties } from "../managementClient";
+import { throwTypeErrorIfParameterMissing, throwErrorIfConnectionClosed } from "../util/error";
+import { getTracer } from "@azure/core-tracing";
+import { SpanContext, Span, SpanKind, CanonicalCode } from "@opentelemetry/types";
+import { getParentSpan } from "../util/operationOptions";
+import { OperationNames } from "../models/private";
+import {
+  GetEventHubPropertiesOptions,
+  GetPartitionIdsOptions,
+  GetPartitionPropertiesOptions,
+  EventHubClientOptions
+} from "../models/public";
+
+export function createAmqpContextWithTokenCredential(
+  host: string,
+  eventHubName: string,
+  credential: TokenCredential,
+  options?: EventHubClientOptions
+): ConnectionContext {
+  if (!eventHubName) {
+    throw new TypeError(`"eventHubName" is missing`);
+  }
+
+  if (!host.endsWith("/")) host += "/";
+  const connectionString = `Endpoint=sb://${host};SharedAccessKeyName=defaultKeyName;SharedAccessKey=defaultKeyValue;EntityPath=${eventHubName}`;
+  const config = EventHubConnectionConfig.create(connectionString);
+
+  ConnectionConfig.validate(config);
+  return ConnectionContext.create(config, credential, options);
+}
+
+export function createAmqpContextUsingConnectionString(
+  connectionString: string,
+  options: EventHubClientOptions
+): ConnectionContext;
+export function createAmqpContextUsingConnectionString(
+  connectionString: string,
+  eventHubName: string,
+  options: EventHubClientOptions
+): ConnectionContext;
+export function createAmqpContextUsingConnectionString(
+  connectionString: string,
+  eventHubNameOrOptions: string | EventHubClientOptions,
+  optionsArg?: EventHubClientOptions
+): ConnectionContext {
+  let config;
+  let credential: SharedKeyCredential;
+  connectionString = String(connectionString);
+
+  let actualEventHubClientOptions: EventHubClientOptions;
+
+  const parsedCS = parseConnectionString<EventHubConnectionStringModel>(connectionString);
+  if (
+    !(parsedCS.EntityPath || (typeof eventHubNameOrOptions === "string" && eventHubNameOrOptions))
+  ) {
+    throw new TypeError(
+      `Either provide "eventHubName" or the "connectionString": "${connectionString}", ` +
+        `must contain "EntityPath=<your-event-hub-name>".`
+    );
+  }
+  if (
+    parsedCS.EntityPath &&
+    typeof eventHubNameOrOptions === "string" &&
+    eventHubNameOrOptions &&
+    parsedCS.EntityPath !== eventHubNameOrOptions
+  ) {
+    throw new TypeError(
+      `The entity path "${parsedCS.EntityPath}" in connectionString: "${connectionString}" ` +
+        `doesn't match with eventHubName: "${eventHubNameOrOptions}".`
+    );
+  }
+  connectionString = connectionString;
```

Comment:
Why?

Reply:
Why what?

Reply:
Looks like a refactoring glitch - the parameter used to be ambiguously typed and then I fixed that, renamed the parameter, and walked away. Leaving one Dwigt.